### PR TITLE
Add utils for processing verifiable presentations

### PR DESCRIPTION
**Pull request by przydatek**
Add to `vc_util` a function `verify_ii_presentation_jwt_with_canister_ids` to verify presentations that are delivered to RP during the attribute sharing flow on the IC.

### CODE REVIEW

File: src/vc_util/src/lib.rs [hunk 1]
```diff
@@ -184,6 +296,10 @@ fn extract_id_alias(claims: &JwtClaims<Value>) -> Result<AliasTuple, JwtValidati
     let id_alias = principal_for_did(alias).map_err(|_| {
         inconsistent_jwt_claims("malformed \"has_id_alias\" claim in id_alias JWT vc")
     })?;
+    println!(
```
Comment (frederikrothenberger) [inline src/vc_util/src/lib.rs:299]: Left over from debugging?

**Review by frederikrothenberger — APPROVED** [review]
I'm not sure it fits this PR, but I think we should tighten the JWS validation, see comments.

Otherwise, LGTM! Thanks.

File: src/vc_util/src/lib.rs [hunk 2]
```diff
@@ -149,6 +165,97 @@ pub fn verify_credential_jws_with_canister_id(
...
+        ));
+    }
+    let id_alias_vc_jws =
+        presentation
+            .verifiable_credential
+            .get(0)
+            .ok_or(PresentationVerificationError::Unknown(
+                "missing id_alias vc".to_string(),
+            ))?;
+    let alias_tuple = get_verified_id_alias_from_jws(
```
[the remark this answers is not in the document]
Comment (przydatek) [reply src/vc_util/src/lib.rs:207]: Agreed, but for this we'd need generation of test-data on-the-fly (or some long-lived test VCs), so I've filed a [ticket](https://dfinity.atlassian.net/browse/GIX-2158) for that.

File: src/vc_util/src/lib.rs [hunk 3]
```diff
@@ -149,6 +165,97 @@ pub fn verify_credential_jws_with_canister_id(
...
+        ));
+    }
+    let requested_vc_jws =
+        presentation
+            .verifiable_credential
+            .get(1)
+            .ok_or(PresentationVerificationError::Unknown(
+                "missing requested vc".to_string(),
+            ))?;
+    let claims = verify_credential_jws_with_canister_id(
```
[the remark this answers is not in the document]
Comment (przydatek) [reply src/vc_util/src/lib.rs:232]: Agreed for `exp`, not so sure for `iss`, as this depends on the context.  I added a note to the ticket.

**Review by przydatek — COMMENTED** [review]
Thanks for the review @frederikrothenberger , will merge when CI is happy.

**Pull request merged**